### ISSUE TITLE
Viewport Problem and using sprite directly

### ISSUE THREAD

**Issue by zhaishengfu** [CaptainProton42/2DGrassShaderDemo#4]
Hello, I found that using viewport had many problems, for example, when i use environment and
 canvasmodulate, or using character camera2D. So I decided to realize your method using sprite directly. Now I almost finished it.

![image](https://user-images.githubusercontent.com/7630101/97942692-3c25c780-1dbf-11eb-9e06-fd2dfdda3d17.png)
I setup the scene as following , without any viewport. 
![image](https://user-images.githubusercontent.com/7630101/97942815-43e56c00-1dbf-11eb-9d4e-bf82ff8a8feb.png)
I realized following fuctions:
1. can generate grass in any positions , not limited by environment and camera2d
2. can dynamically interacted with mouse go generate new height grass
3. can have wind and noise like yours
4. can have the right "grass hide some portion of character" effect

All in all, without viewport I can realize all your functions, and viewport have many limitions. So Why you use viewport and not sprite directly???

**Comment by john-wigg** [comment]
I don't remember the exact reasons I had at the time but I believe I found the viewport to be a bit more flexible initally. It allowed me e.g. to overlap and compose multiple textures inside the viewport. I could also access the viewport texture from any sprite and did not need to worry about positioning since the viewport would always cover the entire screen.

However, I did not really end up making use of these advantages since the final (drawable) texture covered the entire screen anyways. So I guess you're right, it's not really needed here. I tried not putting too much of an emphasis in the tutorial though, since I'm aware that there are advantages and disatvantages to using a viewport.

I will close this later if you don't have any more questions.

EDIT: Wrong account, oops :D

**Comment by zhaishengfu** [comment]
for viewport, i only see disadvantages. you say
'I found the viewport to be a bit more flexible initally' and 
'access the viewport texture from any sprite'
this can be done by sprite directly , considering any scale, pivot and position

and you say 
'the final (drawable) texture covered the entire screen'
this is the problem!. when you have camera2D and the screen is changeable, and you want the grass to be static, the viewport will be big problem. that is , it will change with the Camera2D.

**Comment by CaptainProton42** [comment]
As I said, I just found the viewport to work better for me when I initially started doing this. I still see *some* advantages to using a viewport (such as only having to worry about a single texture at all times). Your approach then is of course completely valid and may be better suited for your needs (especially for Camera2D since I did not really take this into account when making this demo).
